### DISCUSSION TITLE
Does MS GraphRag Send Data to Endpoints Other Than OpenAI or Azure?

### DISCUSSION THREAD

**Discussion by tatsuya-mk1** [microsoft/graphrag#1397] [Q&A]
When using MS GraphRag, does the data I send go exclusively to OpenAI, or is it also sent to other endpoints beyond the OpenAI API in use?

I'm sorry if this question is too basic; I am a beginner, so any clarification would be greatly appreciated.

**Comment by jgbradley1** [comment]
Data is only sent to the OpenAI endpoint that you configure graphrag to use. `graphrag` can be configured to use OpenAI or Azure OpenAI endpoints so it really depends on your setup. `graphrag` can be configured to use other Azure services (Azure Storage, CosmosDB, Azure AI Search, etc.), but only if you explicitly configure it to do so. By default, the graphrag library is setup for anyone to run locally on your computer and will reach out to either an OpenAI or AOAI endpoint for LLM model access.

**Reply by tatsuya-mk1** [reply]
Thank you so much for answering. I am only using the OpenAI endpoint without using Azure srevices so, I guess my worries of data being sent to micrsoft endpoints is unfounded worries.
